src/charm_state.py

```diff
@@ -55,12 +108,27 @@ def from_charm(cls, charm: CharmBase) -> "CharmState":
 
         Return:
             The CharmState instance created by the provided charm.
+
+        Raises:
+            CharmConfigInvalidError: if the charm configuration is invalid.
         """
         keepalive = charm.config.get("webserver_keepalive")
         timeout = charm.config.get("webserver_timeout")
         workers = charm.config.get("webserver_workers")
         threads = charm.config.get("webserver_threads")
+        flask_config = {
+            k.removeprefix("flask_"): v for k, v in charm.config.items() if k.startswith("flask_")
+        }
+        try:
+            valid_flask_config = FlaskConfig(**flask_config)  # type: ignore
+        except ValidationError as exc:
+            error_fields = set(
+                itertools.chain.from_iterable(error["loc"] for error in exc.errors())
+            )
+            error_field_str = " ".join(f"flask_{f}" for f in error_fields)
+            raise CharmConfigInvalidError(f"invalid configuration: {error_field_str}") from exc
         return cls(
+            flask_config=valid_flask_config.dict(exclude_unset=True, exclude_none=True),
```

Comment:
Any specific reason to convert it to a dict ? If FlaskConfig was a dataclass we could use it as is

Reply:
Answered in above.